app/javascript/mastodon/features/account/components/account_note.js

```diff
@@ -92,6 +96,31 @@ class AccountNote extends ImmutablePureComponent {
     this.textarea = c;
   }
 
+  setEditable = () => {
+    const { value } = this.state;
+    const sleep = (waitSeconds) => {
+      return new Promise(resolve => {
+        setTimeout(() => {
+          resolve();
+        }, waitSeconds);
+      });
+    };
+    let my = this;
+    sleep(50)
+      .then(() => {
+        my.setState({ editable: true });
+        my.textarea.focus();
+        const len = value.length;
+        my.textarea.setSelectionRange(len, len);
+      }).catch(() => {
+        my.setState({ editable: false });
+      });
```

Comment:
I'm not sure if/how much an issue is the textarea being focused at the start of the text rather than at the end. If you do keep that behavior, I'd just write this function with `setTimeout(() => { my.setState(…); …)` rather than using a Promise there.